packages/theme-language-server-common/src/completions/providers/FilterCompletionProvider.spec.ts

```diff
@@ -133,6 +168,42 @@ describe('Module: FilterCompletionProvider', async () => {
     // As in, the anyFilters are at the _end_ and not shown at the top.
     await expect(provider).to.complete('{{ string | █ }}', stringFilters.concat(anyFilters));
   });
+
+  describe('when there are no required parameters', () => {
+    it('should not include parameters in the insertText of the completion', async () => {
+      await expect(provider).to.complete('{{ string | defa█ }}', [
+        expect.objectContaining({
+          label: 'default',
+          insertText: 'default',
+          insertTextFormat: 1,
+        }),
+      ]);
+    });
+  });
+
+  describe('when there are required positional parameters', () => {
+    it('should include parameters in the insertText of the completion', async () => {
+      await expect(provider).to.complete('{{ string | h█ }}', [
+        expect.objectContaining({
+          label: 'highlight',
+          insertText: "highlight: '${1:highlighted_term}'",
+          insertTextFormat: 2,
+        }),
+      ]);
+    });
+  });
+
+  describe('when there are required named parameters', () => {
+    it('should include parameters in the insertText of the completion', async () => {
+      await expect(provider).to.complete('{{ string | pre█ }}', [
```

Comment:
Can we add a test to make sure the required args are not added if text follows?

I typically use `TextEdit` instead of `insertText` in order to maybe replace stuff past the cursor. It might not apply here, but we probably don't want `width: 100, height: 100` to be appended the second time you accept the image_url filter 🤔

Reply:
I should have added this to the PR description that I'm aware of this bug but specifically not addressing it here. I didn't want to balloon this PR any more and it's an outstanding bug. For example, even on `main` we get the duplicate filter:

![bug](https://github.com/user-attachments/assets/c232b17c-abec-45b9-9bd7-73a297893f0d)

This change does make it *worse*, though:

![badbug](https://github.com/user-attachments/assets/ba11b30a-0d35-4caf-8e38-0b892678fe49)

Reply:
I wonder if I can abstract https://github.com/Shopify/theme-tools/pull/522/files#diff-7256ee5482c911787e4800568fe9ea339c55f5a946fa72793694f4b2cdfe3b9aR70 into some shared code 🤔 

I'll have a look at this

Reply:
This was hard! https://github.com/Shopify/theme-tools/pull/522/commits/4b2dec31b946b8e92d8601390699c0af94d0274c